src/Microsoft.DotNet.CodeFormatter.Analyzers.Tests/ExplicitVariableTypeAnalyzerTests.cs

```diff
@@ -0,0 +1,399 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.DotNet.CodeFormatter.Analyzers;
+
+using Xunit;
+
+namespace Microsoft.DotNet.CodeFormatter.Analyzers.Tests
+{
+    public sealed class ExplicitVariableTypeAnalyzerTests : AnalyzerFixerTestBase
+    {
+        public ExplicitVariableTypeAnalyzerTests()
+        {
+            DisableAllDiagnostics();
+            EnableDiagnostic(ExplicitVariableTypeAnalyzer.DiagnosticId);
+        }
+
+        [Fact]
+        public void TestSimpleTypeExplicitDeclaration()
+        {
+            const string text = @"
+class C1
+{
+    void M()
+    {
+        int x = 0;
+        bool[] y = { true };  
+        int[][] z = new[] { new[] { 1, 2, 3 }, new[] { 4, 5, 6 } };
+    }
+}"; 
+            Verify(text, text, runFormatter: false);
+        }
+
+        [Fact]
+        public void TestSimpleObviousTypeVarDeclaration()
+        {
+            const string text = @"
```

Comment:
consider using 'input' as a name instead of text. also suggest always creating an 'expected' string variable. set expected = input; where that makes sense. alternately you could name your parameters in your verify call, like so:

Verify(input, expected : input, runFormatter: false)

I think the former would be better. this is a nit. your tests are very nice